File: app/src/main/java/com/gmail/fattazzo/formula1world/fragments/current/races/detail/pages/statistics/pitstops/RaceStatPitStopItemView.java
package com.gmail.fattazzo.formula1world.fragments.current.races.detail.pages.statistics.pitstops;

import android.content.Context;
import android.util.TypedValue;
import android.widget.TableRow;
import android.widget.TextView;

import com.gmail.fattazzo.formula1world.R;
import com.gmail.fattazzo.formula1world.domain.F1PitStop;

import org.androidannotations.annotations.AfterViews;
import org.androidannotations.annotations.EViewGroup;
import org.androidannotations.annotations.ViewById;

/**
 * @author fattazzo
 *         <p/>
 *         date: 20/04/17
 */
@EViewGroup(R.layout.race_stat_pitstops_row)
public class RaceStatPitStopItemView extends TableRow {

    @ViewById
    TextView driver_tv;

    @ViewById
    TextView stop_tv;

    @ViewById
    TextView lap_tv;

    @ViewById
    TextView time_tv;

    @ViewById
    TextView duration_tv;

    private F1PitStop pitStop;

    private int rowNumber;

    public RaceStatPitStopItemView(Context context, F1PitStop pitStop, int rowNumber) {
        super(context);
        this.pitStop = pitStop;
        this.rowNumber = rowNumber;
    }

    @AfterViews
    void bind() {
        driver_tv.setText(pitStop.driver.getFullName());
        stop_tv.setText(String.valueOf(pitStop.stop));
        lap_tv.setText(String.valueOf(pitStop.lap));
        time_tv.setText(pitStop.time);
        duration_tv.setText(pitStop.duration);

        setBackgroundColor(rowNumber % 2 == 0 ? getThemeEvenRowColor() : getThemeOddRowColor());
    }

    private int getThemeEvenRowColor() {
        final TypedValue value = new TypedValue();
        getContext().getTheme().resolveAttribute(R.attr.evenRowColor, value, true);
        return value.data;
    }

    private int getThemeOddRowColor() {
        final TypedValue value = new TypedValue();
        getContext().getTheme().resolveAttribute(R.attr.oddRowColor, value, true);
        return value.data;
    }
}

File: app/src/main/kotlin/com/gmail/fattazzo/formula1world/fragments/current/races/detail/pages/statistics/pitstops/RaceStatPitStopItemView.kt
/*
 * Project: total-gp-world
 * File: RaceStatPitStopItemView.kt
 *
 * Created by fattazzo
 * Copyright © 2018 Gianluca Fattarsi. All rights reserved.
 *
 * MIT License
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */

package com.gmail.fattazzo.formula1world.fragments.current.races.detail.pages.statistics.pitstops

import android.content.Context
import android.util.TypedValue
import android.widget.TableRow
import android.widget.TextView

import com.gmail.fattazzo.formula1world.R
import com.gmail.fattazzo.formula1world.domain.F1PitStop

import org.androidannotations.annotations.AfterViews
import org.androidannotations.annotations.EViewGroup
import org.androidannotations.annotations.ViewById

/**
 * @author fattazzo
 *
 *
 * date: 20/04/17
 */
@EViewGroup(R.layout.race_stat_pitstops_row)
open class RaceStatPitStopItemView(context: Context, private val pitStop: F1PitStop, private val rowNumber: Int) : TableRow(context) {

    @ViewById
    lateinit internal var driver_tv: TextView

    @ViewById
    lateinit internal var stop_tv: TextView

    @ViewById
    lateinit internal var lap_tv: TextView

    @ViewById
    lateinit internal var time_tv: TextView

    @ViewById
    lateinit internal var duration_tv: TextView

    private val themeEvenRowColor: Int
        get() {
            val value = TypedValue()
            context.theme.resolveAttribute(R.attr.evenRowColor, value, true)
            return value.data
        }

    private val themeOddRowColor: Int
        get() {
            val value = TypedValue()
            context.theme.resolveAttribute(R.attr.oddRowColor, value, true)
            return value.data
        }

    @AfterViews
    internal fun bind() {
        driver_tv!!.text = pitStop.driver!!.fullName
        stop_tv!!.text = pitStop.stop.toString()
        lap_tv!!.text = pitStop.lap.toString()
        time_tv!!.text = pitStop.time
        duration_tv!!.text = pitStop.duration

        setBackgroundColor(if (rowNumber % 2 == 0) themeEvenRowColor else themeOddRowColor)
    }
}
